pkg/console/tlsconfig/tlsconfig.go

```diff
@@ -0,0 +1,281 @@
+		return nil, w.certError
+	}
+
+	if w.certificate == nil {
+		return nil, fmt.Errorf("certificate not loaded")
+	}
+
+	return &tls.Config{
+		CipherSuites: w.ciphers,
+		MinVersion:   w.minTlsVersion,
+		Certificates: []tls.Certificate{*w.certificate},
+	}, nil
+}
+
+func (w *watch) processEvents(watcher *fsnotify.Watcher, done <-chan struct{}) error {
+	w.running.Store(true)
+	defer w.running.Store(false)
+
+	for {
+		select {
+		case <-done:
+			return nil
+
+		case event, ok := <-watcher.Events:
+			if !ok {
+				return nil
+			}
+			w.handleEvent(event)
+
+		case err, ok := <-watcher.Errors:
+			if !ok {
+				return nil
+			}
+			if err != nil {
+				return err
+			}
+		}
+	}
+}
+
+func (w *watch) handleEvent(event fsnotify.Event) {
+	const modificationEvents = fsnotify.Create | fsnotify.Write | fsnotify.Remove
+	if event.Op&modificationEvents == 0 {
+		return
+	}
+
+	switch {
+	case strings.HasPrefix(w.tlsProfilePath, event.Name):
+		w.reloadTlsProfile()
+	case strings.HasPrefix(w.certsPath, event.Name):
+		w.reloadCertificate()
+	case strings.HasPrefix(w.keyPath, event.Name):
+		w.reloadCertificate()
+	}
+}
+
+func (w *watch) reloadTlsProfile() {
+	w.lock.Lock()
+	defer w.lock.Unlock()
+	w.tlsProfileError = nil
+
+	ciphers, minVersion, err := loadCipherSuitesAndMinVersion(w.tlsProfilePath)
+	if errors.Is(err, os.ErrNotExist) {
+		// Config file does not exist, using zero values for default
+		w.ciphers = nil
+		w.minTlsVersion = 0
+		return
+	}
+	if err != nil {
+		log.Log.Errorf("Failed to load TLS configuration: %s", err)
+		w.tlsProfileError = fmt.Errorf("failed to load TLS configuration: %w", err)
+		return
+	}
+
+	log.Log.Infof("Loaded TLS configuration.")
+	w.ciphers = ciphers
+	w.minTlsVersion = minVersion
+}
+
+func (w *watch) reloadCertificate() {
+	w.lock.Lock()
+	defer w.lock.Unlock()
+	w.certError = nil
+
+	certificate, err := LoadCertificates(w.certsPath, w.keyPath)
+	if err != nil {
+		w.certError = err
+		return
+	}
+
+	log.Log.Infof("Loaded TLS certificate.")
+	w.certificate = certificate
+}
+
+func loadCipherSuitesAndMinVersion(configPath string) ([]uint16, uint16, error) {
+	tlsProfile, err := loadTlsProfile(configPath)
+	if err != nil {
+		return nil, 0, fmt.Errorf("could not load tls config: %w", err)
+	}
+
+	tlsProfileSpec, err := getTlsProfileSpec(tlsProfile)
+	if err != nil {
+		return nil, 0, err
+	}
+
+	ciphers := getCipherSuites(tlsProfileSpec)
+	minVersion, err := crypto.TLSVersion(string(tlsProfileSpec.MinTLSVersion))
+	if err != nil {
+		return nil, 0, err
+	}
+
+	return ciphers, minVersion, nil
+}
+
+func loadTlsProfile(profilePath string) (*v1alpha1.TlsSecurityProfile, error) {
+	file, err := os.Open(profilePath)
+	if err != nil {
+		return nil, fmt.Errorf("error opening file: %w", err)
+	}
+	// It's ok to ignore error on close, because the file is opened of reading
+	defer func() { _ = file.Close() }()
+
+	result := &v1alpha1.TlsSecurityProfile{}
+	err = yaml.NewYAMLOrJSONDecoder(file, 1024).Decode(result)
```

Comment:
Better settle for one format only (YAML)?

Reply:
done